### PR TITLE
Feature 73 on call calendar thesis work

### PR DESCRIPTION
**Pull request by petri-nykanen**
This PR introduces the on-call calendar UI.

- Calendar view
- List view (old on-call calendar)
- On-call card in the home screen

### CODE REVIEW

File: src/atoms/oncall.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import { atom } from "jotai";
+import type { OnCall } from "src/generated/homeLambdasClient";
+
+export const oncallAtom = atom<OnCall[]>([]);
```
Comment (Jdallos) [inline src/atoms/oncall.ts:4]:
```suggestion
export const onCallAtom = atom<OnCall[]>([]);
```

File: src/components/contexts/oncall-handler.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+/**
+ * On call handler component
+ * @param props component properties
+ */
+const OnCallHandler = ({ open, setOpen, onCallEntry, updatePaidStatus }: Props) => {
+  const [confirmationOpen, setConfirmationOpen] = useState(false);
+  const weekNumber = DateTime.fromISO(String(onCallEntry?.date)).weekNumber;
+  const year = DateTime.fromISO(String(onCallEntry?.date)).year;
+
+  if (onCallEntry)
```
Comment (Jdallos) [inline src/components/contexts/oncall-handler.tsx:27]: For clarity change to if !onCallEntry return null

File: src/components/contexts/oncall-handler.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    /**
+     * Component render
+     */
+    return (
+      <GenericDialog
+        title={`${strings.timeExpressions.week} ${
```
Comment (Jdallos) [inline src/components/contexts/oncall-handler.tsx:33]: Create a named variable for this string- to improve code clarity

File: src/components/contexts/oncall-handler.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+          DateTime.fromISO(String(onCallEntry.date)).weekNumber
+        }`}
+        open={open}
+        error={false}
```
Comment (Jdallos) [inline src/components/contexts/oncall-handler.tsx:37]: Never true? If not is this needed?

File: src/components/screens/on-call-calendar-screen.tsx [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+   * Fetches on call data, such as personnel and weeks
+   * @param year year of entries to search
+   */
+  const getOnCallData = async (year: number) => {
+    setIsLoading(true);
+    try {
+      const fetchedData = await onCallApi.listOnCallData({ year: year.toString() });
+      setOnCallData(fetchedData);
+    } catch (error) {
+      setError(`${strings.error.fetchFailedGeneral}, ${error}`);
```
Comment (Jdallos) [inline src/components/screens/on-call-calendar-screen.tsx:68]: More specific error message

File: src/components/home/oncall-card.tsx [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    getOnCallData(DateTime.now().year);
+  }, []);
+
+  const getOnCallData = async (year: number) => {
+    try {
+      const fetchedData = await onCallApi.listOnCallData({ year: year.toString() });
+      setOnCallData(fetchedData);
+      getCurrentOnCallPerson();
+    } catch (error) {
+      setError(`${strings.error.fetchFailedGeneral}, ${error}`);
```
Comment (Jdallos) [inline src/components/home/oncall-card.tsx:27]: Specific error message

File: src/components/home/oncall-card.tsx [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+  const { onCallApi } = useLambdasApi();
+  const [onCallData, setOnCallData] = useAtom(oncallAtom);
+
+  const setError = useSetAtom(errorAtom);
+
+  useEffect(() => {
+    getOnCallData(DateTime.now().year);
+  }, []);
+
+  const getOnCallData = async (year: number) => {
```
Comment (Jdallos) [inline src/components/home/oncall-card.tsx:21]: Docs

File: src/components/home/oncall-card.tsx [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+import { useAtom, useSetAtom } from "jotai";
+import { DateTime } from "luxon";
+import { useEffect } from "react";
+import { Link } from "react-router-dom";
+import { errorAtom } from "src/atoms/error";
+import { oncallAtom } from "src/atoms/oncall";
+import { useLambdasApi } from "src/hooks/use-api";
+import strings from "src/localization/strings";
+
+const OnCallCard = () => {
```
Comment (Jdallos) [inline src/components/home/oncall-card.tsx:11]: Docs

File: src/components/home/oncall-card.tsx [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+    try {
+      const fetchedData = await onCallApi.listOnCallData({ year: year.toString() });
+      setOnCallData(fetchedData);
+      getCurrentOnCallPerson();
+    } catch (error) {
+      setError(`${strings.error.fetchFailedGeneral}, ${error}`);
+    }
+  };
+
+  const getCurrentOnCallPerson = () => {
```
Comment (Jdallos) [inline src/components/home/oncall-card.tsx:31]: Docs

**Review by Jdallos — CHANGES_REQUESTED** [review]

File: src/components/screens/on-call-calendar-screen.tsx [hunk 10]
```diff
@@ -0,0 +1,259 @@
...
+  const [onCallData, setOnCallData] = useAtom(onCallAtom);
+  const [open, setOpen] = useState(false);
+  const [isCalendarView, setIsCalendarView] = useState(validateJSONString());
+  const [selectedDate, setSelectedDate] = useState<DateTime>(DateTime.now());
+  const [onCallPerson, setOnCallPerson] = useState("");
+  const [selectedOnCallEntry, setSelectedOnCallEntry] = useState<OnCallCalendarEntry>();
+  const setError = useSetAtom(errorAtom);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
```
Comment (McDuuck) [inline src/components/screens/on-call-calendar-screen.tsx:49]: After year changes this useEffect gives error in the UI

Comment (McDuuck) [reply src/components/screens/on-call-calendar-screen.tsx:49]: GET "URL" 400 (Bad Request)

**Pull request closed without merging**